### PR TITLE
ci: Combine release jobs, add release approval

### PR DESCRIPTION
**Pull request by mslipper**
Combines the Docker build/publish jobs into one, and makes them all depend on a `hold` job that asks for approval. The `hold` step allows someone from the `release-managers` security group to approve the build before it is released. Without this step, tags created by automation tools like GitHub Actions will not be pushed.

I also added automatic release tags to each Docker build.

### CODE REVIEW

**Comment by wecreateproject** [comment]
.

File: .circleci/config.yml [hunk 1]
```diff
@@ -199,17 +213,36 @@ jobs:
...
+    machine:
+      image: ubuntu-2204:2022.07.1
+      resource_class: xlarge
     steps:
       - gcp-cli/install
       - gcp-oidc-authenticate
       - checkout
       - run:
-          name: Tag
+          name: Build
```
Comment (tynes) [inline .circleci/config.yml:224]: I'm fine with this being inlined but also may be better to have more complex logic in a script and pass in args

**Review by tynes — APPROVED** [review]
Generally looks good to me altho complex enough to need to see it run and work to be sure

File: .circleci/config.yml (hunk 1 above)
Comment (mslipper) [reply .circleci/config.yml:224]: I can clean this up in separate PRs - need to see it run against develop first.

**Comment by mslipper** [comment]
I need to merge this to develop to actually test it - the OIDC permissions require a full GCP account to be set up. I'll run a test now, and push bugfixes as necessary.

**Pull request merged**